hvplot/__init__.py

```diff
@@ -55,21 +55,36 @@ def _get_doc(cls, kind, completions=False, docstring=True, generic=True, style=T
         defaults = argspec.defaults or [None]*len(args)
         for arg, dflt in zip(args, defaults):
             parameters.append((arg, dflt))
+        signature = None
     else:
         sig = inspect.signature(method)
         for name, p in list(sig.parameters.items())[1:]:
             if p.kind == 1:
                 parameters.append((name, p.default))
+
+        extra_kwargs = _hv.core.util.unique_iterator(valid_opts + kind_opts
+                                                     + converter._axis_options
+                                                     + converter._op_options)
+        filtered_signature = [p for p in sig.parameters.values()
+                              if p.kind != inspect.Parameter.VAR_KEYWORD]
+        extra_params = [inspect.Parameter(k, inspect.Parameter.KEYWORD_ONLY)
+                        for k in extra_kwargs
+                        if k not in [p.name for p in filtered_signature]]
+        all_params = (filtered_signature + extra_params
+                      + [inspect.Parameter('kwargs', inspect.Parameter.VAR_KEYWORD)])
```

Comment:
Note that the ``**kwargs`` `VAR_KEYWORD`` at the end here is not strictly needed for tab-completion to work. If we don't except any other ``kwargs`` I am happy to remove it.